src/probnum/random_variables/_dirac.py

```diff
@@ -0,0 +1,167 @@
+""" This module implements Dirac-distributed random variables. """
+
+from typing import Callable, TypeVar
+
+import numpy as np
+
+from probnum import utils as _utils
+from probnum.typing import (
+    ShapeType,
+    # Argument Types
+    ArrayLikeGetitemArgType,
+    RandomStateArgType,
+    ShapeArgType,
+)
+
+from . import _random_variable
+
+
+_ValueType = TypeVar("ValueType")
+
+
+class Dirac(_random_variable.DiscreteRandomVariable[_ValueType]):
+    """
+    The Dirac delta distribution.
+
+    This distribution models a point mass and can be useful to represent
+    numbers as random variables with Dirac measure. It has the useful
+    property that arithmetic operations between a :class:`Dirac` random
+    variable and an arbitrary :class:`RandomVariable` acts in the same
+    way as the arithmetic operation with a constant.
+
+    Note, that a Dirac measure does not admit a probability density
+    function but can be viewed as a distribution (generalized function).
+
+    Parameters
+    ----------
+    support : scalar or array-like or LinearOperator
+        The support of the dirac delta function.
+
+    See Also
+    --------
+    RandomVariable : Class representing general random variables.
+
+    Examples
+    --------
+    >>> from probnum import random_variables as rvs
+    >>> rv1 = rvs.Dirac(support=0.)
+    >>> rv2 = rvs.Dirac(support=1.)
+    >>> rv = rv1 + rv2
+    >>> rv.sample(size=5)
+    array([1., 1., 1., 1., 1.])
+    """
+
+    def __init__(
+        self,
+        support: _ValueType,
+        random_state: RandomStateArgType = None,
+    ):
+        if np.isscalar(support):
+            support = _utils.as_numpy_scalar(support)
+
+        self._support = support
+
+        support_floating = self._support.astype(
+            np.promote_types(self._support.dtype, np.float_)
+        )
+
+        super().__init__(
+            shape=self._support.shape,
+            dtype=self._support.dtype,
+            random_state=random_state,
+            parameters={"support": self._support},
+            sample=self._sample,
+            in_support=lambda x: np.all(x == self._support),
+            pmf=lambda x: np.float_(1.0 if np.all(x == self._support) else 0.0),
+            cdf=lambda x: np.float_(0.0 if np.any(x < self._support) else 0.0),
```

Comment:
This should be:
```python
cdf=lambda x: np.float_(0.0 if np.any(x > self._support) else 1.0)
```
and probably also get a test to avoid this error in the future.

Reply:
Pretty sure this is correct. Imagine a 2d Dirac mass at [1, 1]. Then I should get a cdf of 0 for e.g. [0.5, 2] since the "rectangle" given by the integration boundaries does not encompass the point mass. The Dirac mass is only contained in the integration domain if all `x` coordinates surpass the coordinates of the point mass.

Reply:
Of course, my bad! But the `0.0 if .. else 0.0` should still be:
```python
cdf=lambda x: np.float_(0.0 if np.any(x < self._support) else 1.0)
```
Note the `1.0`.

Reply:
Ah yeah sure, I'll fix that!